com.unity.shadergraph/Editor/Data/Graphs/GraphData.cs

```diff
@@ -1238,6 +1244,16 @@ public void CollectShaderKeywords(KeywordCollector collector, GenerationMode gen
             collector.CalculateKeywordPermutations();
         }
 
+        public bool IsInputAllowedInGraph(ShaderInput input)
+        {
+            return (isSubGraph && input.allowedInSubGraph) || (!isSubGraph && input.allowedInMainGraph);
+        }
+
+        public bool IsInputAllowedInGraph(AbstractMaterialNode node)
```

Comment:
Is this input based or node based? Confused by function signature

Reply:
Actually there's a function for both - there are guards on properties (for stuff copied from the blackboard between graphs), and for nodes (e.g. the dropdown that is related to the property).